src/recipe_dictionary.h

```diff
@@ -0,0 +1,70 @@
+#ifndef RECIPE_DICTIONARY_H
+#define RECIPE_DICTIONARY_H
+
+#include "crafting.h" // for recipe, FIXME: circular dependency -> possible fix: move recipe to this file?
```

Comment:
If you only have pointers to recipes in the class definition, the ```struct recipe;``` declaration below will be sufficient and you don't need an include of the full recipe class definition.